src/python/graplctl/graplctl/aws/lib.py

```diff
@@ -18,108 +16,32 @@
-
-    if edge_ux_artifact_dir.exists():
-        shutil.rmtree(edge_ux_artifact_dir)
-
-    os.mkdir(edge_ux_artifact_dir)
-
-    LOGGER.info("building cdk")
-    _run_shell_cmd("npm run build", cwd=grapl_cdk_dir, stdout=stdout, stderr=stderr)
-    LOGGER.info("built cdk")
-
-    LOGGER.info("deploying Grapl stack")
-    _run_shell_cmd(
-        f"cdk deploy --require-approval=never --profile={aws_profile} --outputs-file={outputs_file.as_posix()} Grapl",
-        cwd=grapl_cdk_dir,
-        stdout=stdout,
-        stderr=stderr,
-    )
-    LOGGER.info("deployed Grapl stack")
-
-    shutil.rmtree(edge_ux_artifact_dir)
-    os.mkdir(edge_ux_artifact_dir)
-
-    LOGGER.info("creating edge UX package")
-    _run_shell_cmd(
-        "npm run create_edge_ux_package",
-        cwd=grapl_cdk_dir,
-        stdout=stdout,
-        stderr=stderr,
-    )
-    LOGGER.info("created edge UX package")
-
-    LOGGER.info("deploying EngagementUX stack")
-    _run_shell_cmd(
-        f"cdk deploy --require-approval=never --profile={aws_profile} --outputs-file={outputs_file.as_posix()} EngagementUX",
-        cwd=grapl_cdk_dir,
-        stdout=stdout,
-        stderr=stderr,
-    )
-    LOGGER.info("deployed EngagementUX stack")
-
-    shutil.rmtree(edge_ux_artifact_dir)
-
-
-def destroy_grapl(
-    grapl_root: pathlib.Path, aws_profile: str, stdout: IO[AnyStr], stderr: IO[AnyStr]
-) -> None:
-    grapl_cdk_dir = pathlib.Path(grapl_root.absolute(), GRAPL_CDK_RELATIVE_PATH)
-
-    LOGGER.info("building cdk")
-    _run_shell_cmd("npm run build", cwd=grapl_cdk_dir, stdout=stdout, stderr=stderr)
-    LOGGER.info("built cdk")
-
-    LOGGER.info("destroying all stacks")
-    _run_shell_cmd(
-        f'cdk destroy --profile={aws_profile} --force --require-approval=never "*"',
-        cwd=grapl_cdk_dir,
-        stdout=stdout,
-        stderr=stderr,
-    )
-    LOGGER.info("destroyed all stacks")
-
-
-def provision_grapl(lambda_: LambdaClient, deployment_name: str) -> None:
-    LOGGER.info("invoking provisioner lambda")
+def _invoke_lambda(lambda_: LambdaClient, function_name: str) -> None:
+    LOGGER.info(f"invoking lambda {function_name}")
     result = lambda_.invoke(
-        FunctionName=f"{deployment_name}-Provisioner-Handler",
+        FunctionName=function_name,
         InvocationType="RequestResponse",
         LogType="Tail",
     )
 
     status = result["StatusCode"]
     logs = base64.b64decode(bytes(result["LogResult"], "utf-8")).decode("utf-8")
-    if status == 200:
+    if status == 200 and result.get("FunctionError") is None:
         for line in logs.splitlines():
             LOGGER.info(line)
-        LOGGER.info("provisioner lambda succeeded")
+        LOGGER.info(f"lambda invocation succeeded for {function_name}")
     else:
-        for line in logs.splitlines():
-            LOGGER.error(line)
+        LOGGER.error(
+            f"{''.join(l.decode('utf-8') for l in cast(StreamingBody, result['Payload']).iter_lines())}"
```

Comment:
1. could we, like, bring this out to a separate function or something
2. looks like the f-string is no longer required